src/main/java/teammates/ui/controller/CreateImageUploadUrlAction.java

```diff
@@ -0,0 +1,40 @@
+package teammates.ui.controller;
+
+import com.google.appengine.api.blobstore.BlobstoreFailureException;
+
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.util.Const;
+import teammates.common.util.GoogleCloudStorageHelper;
+
+public class CreateImageUploadUrlAction extends Action {
+
+    @Override
+    protected ActionResult execute() throws EntityDoesNotExistException {
+        return createAjaxResult(createImageUploadUrlPageData());
+    }
+
+    protected final CreateImageUploadUrlAjaxPageData createImageUploadUrlPageData() {
+        verifyPrivileges();
+        CreateImageUploadUrlAjaxPageData data = new CreateImageUploadUrlAjaxPageData(account);
+
+        try {
+            data.nextUploadUrl = getUploadUrl();
+            data.ajaxStatus = "Image upload url created, proceed to uploading";
+        } catch (BlobstoreFailureException | IllegalArgumentException e) {
+            data.nextUploadUrl = null;
+            isError = true;
+            data.ajaxStatus = "An error occurred when creating upload URL, please try again";
+        }
+
+        return data;
+    }
+
+    protected void verifyPrivileges() {
```

Comment:
As I understand CreateImageUploadUrlAction can be instantiated by ActionFactory via reflection.
So it can't be abstract and as a result I can't create an abstract method in a non-abstract class